### PR TITLE
Use live bidding with AppBidding

### PR DESCRIPTION
**Pull request by TurpIF**
JIRa: EE-1224

### CODE REVIEW

**Comment by Amokrane** [comment]
Few nitpicks regarding commit messages. The first one, in particular, can be a bit more concise.

**Commit#1**:
The interstitial format is not available if the
`CriteoInterstitialActivity` activity is ~~only~~ not registered correctly in the
Android manifest.

This occurred ~~before~~ when publishers w~h~ere using distributed AAR files.
Now that the SDK is distributed only through Maven repositories, such
unavailability should not happen.

This is not totally verified and checking this via remote logs could
determine~d~ if it is a useless check.

~Anyway it is expected to almost never happens~ (already mentioned above). This commit might make
Criteo bids for nothing (what does bidding for nothing means?). Although, the purpose of this is to make the
`Criteo#loadBid` API shar**e**able ~~for~~ between InHouse and AppBidding (after Although..., I expect an explanation. Eg: Although I am tired, I will continue fighting to the end). ~~So~~ Therefore, anything
related to InHouse should be moved from that call.

**Commit#2**:
This is to make the `loadBid` method shar**e**able between both InHouse and
AppBidding.


**Commit#4**:
This component ~~has no more to do~~ has nothing to with InHouse integration anymore. Instead, it
is just managing the bid served directly to publishers. The
particularity of such bid objects is that they are consumable and valid
only once.

**Comment by TurpIF** [comment]
> Few nitpicks regarding commit messages. The first one, in particular, can be a bit more concise.
> 
> **Commit#1**:
> 
> ```
> The interstitial format is not available if the
> `CriteoInterstitialActivity` activity is ~~only~~ not registered correctly in the
> Android manifest.
> This occurred ~~before~~ when publishers w~h~ere using distributed AAR files.
> Now that the SDK is distributed only through Maven repositories, such
> unavailability should not happen.
> 
> This is not totally verified and checking this via remote logs could
> determine~d~ if it is a useless check.
> 
> ~Anyway it is expected to almost never happens~ (already mentioned above). This commit might make
> Criteo bids for nothing (what does bidding for nothing means?). Although, the purpose of this is to make the
> `Criteo#loadBid` API shar**e**able ~~for~~ between InHouse and AppBidding (after Although..., I expect an explanation. Eg: Although I am tired, I will continue fighting to the end). ~~So~~ Therefore, anything
> related to InHouse should be moved from that call.
> ```
> 
> The last two sentences can probably be rewritten into something more concise (straight to the point, avoid repetitions of ideas, remove useless words).
> 
> **Commit#2**:
> 
> ```
> This is to make the `loadBid` method shar**e**able between both InHouse and
> AppBidding.
> ```
> 
> **Commit#4**:
> 
> ```
> This component ~~has no more to do~~ has nothing to with InHouse integration anymore. Instead, it
> is just managing the bid served directly to publishers. The
> particularity of such bid objects is that they are consumable and valid
> only once.
> ```

Done

File: publisher-sdk/src/main/java/com/criteo/publisher/ConsumableBidLoader.java [hunk 1]
```diff
@@ -21,15 +21,15 @@
 import com.criteo.publisher.model.CdbResponseSlot;
 import org.jetbrains.annotations.NotNull;
 
-public class InHouse {
+public class ConsumableBidLoader {
```
Comment (Amokrane) [inline publisher-sdk/src/main/java/com/criteo/publisher/ConsumableBidLoader.java:24]: I think it's worth documenting why this class has such a name (even if the class is internal). Today we interpret that `consuming` as: being able to use the `Bid` only once. But, if another person comes in this won't be entirely clear and she will think: what does an unconsumable bid mean?

**Review by Amokrane — COMMENTED** [review]
Are there tests that cover the time-budget logic for `AppBidding` (similar question for `InHouse`)? Or do we rely on the fact that BidManager & whatever is below it is fully tested?

File: publisher-sdk/src/main/java/com/criteo/publisher/ConsumableBidLoader.java (hunk 1 above)
Comment (TurpIF) [reply publisher-sdk/src/main/java/com/criteo/publisher/ConsumableBidLoader.java:30]: Done

File: publisher-sdk/src/main/java/com/criteo/publisher/CriteoInternal.java [hunk 2]
```diff
@@ -119,21 +119,20 @@ public void runSafely() {
   }
 
   @Override
-  public void setBidsForAdUnit(Object object, @NonNull AdUnit adUnit) {
+  public void setBidsForAdUnit(Object object, @Nullable Bid bid) {
```
[the remark this answers is not in the document]
Comment (TurpIF) [reply publisher-sdk/src/main/java/com/criteo/publisher/CriteoInternal.java]: Indeed, I almost forgot this.
I added a commit with the renaming.

**Comment by TurpIF** [comment]
> Are there tests that cover the time-budget logic for `AppBidding` (similar question for `InHouse`)? Or do we rely on the fact that BidManager & whatever is below it is fully tested ?

For the moment we strongly rely on the fact that BidManager is fully tested and AppBidding & InHouse are unit tested.
However, I've just ran the test with live bidding enabled, and bid is impossible: we forgot to set the time of download when live bidding is on.
So I would like to add, at least, tests validating that a bid is possible.

As you prefer, I can either add tests right now with the fix of the time of download, or push a new PR with tests+fix

**Comment by Amokrane** [comment]
> As you prefer, I can either add tests right now with the fix of the time of download, or push a new PR with tests+fix

Whatever makes to you is fine by me :)

**Review by Amokrane — APPROVED** [review]

**Comment by TurpIF** [comment]
> > As you prefer, I can either add tests right now with the fix of the time of download, or push a new PR with tests+fix
> 
> Whatever makes sense to you is fine by me :) I have a slight preference for a new PR.

I agree, this one is big enough. I will make a new one.

**Pull request merged**